### PR TITLE
New models

### PR DESCRIPTION
**Pull request by AslonBadridinovGH**

### CODE REVIEW

File: frontend/src/AddNewBook.tsx [hunk 1]
```diff
@@ -10,42 +11,111 @@ import {ChangeEvent, FormEvent, useState} from "react";
 export default function AddNewBook(props : AddNewBookProps){
 
     const [title, setTitle] = useState<string>("")
+
     const [author, setAuthor] = useState<string>("")
 
+    const [genre, setGenre] = useState<string>("")
+
+    const [year, setYears] = useState<string>("")
```
Comment (AslonBadridinovGH) [inline frontend/src/AddNewBook.tsx:19]: change string to number

File: frontend/src/components/edit-book.tsx [hunk 2]
```diff
@@ -24,24 +39,56 @@ export const EditBook: React.FC<EditBookProps> = ({books, editBook}) => {
...
             id: book?.id || "",
-            title, author
+            title, author, genre, year, publisher, city,page,description,views
         })
 
         navigate(`/books/${book?.id}`)
     }
 
+    const redirect = ()=>{
+        navigate("/")
```
Comment (AslonBadridinovGH) [inline frontend/src/components/edit-book.tsx:84]: redirect to details page

File: frontend/src/components/edit-book.tsx [hunk 3]
```diff
@@ -51,7 +98,45 @@ export const EditBook: React.FC<EditBookProps> = ({books, editBook}) => {
...
+                        <div>Description</div>
+                        <input name={"description"} value={description} onChange={onDescriptionChange}
+                               placeholder="Description..."/>
+                    </div>
+                    <div>
+                        <div>Views</div>
+                        <input name={"views"} value={views} onChange={onViewsChange} placeholder="Views..."/>
+                    </div>
+
                     <button type="submit">Edit book</button>
```
Comment (AslonBadridinovGH) [inline frontend/src/components/edit-book.tsx:138]: put both  to one button

File: frontend/src/types/Book.ts [hunk 4]
```diff
@@ -3,5 +3,13 @@ export type Book = {
     id: string;
     title: string;
     author: string;
+    "genre": string,
+    "year": string,
```
Comment (AslonBadridinovGH) [inline frontend/src/types/Book.ts:7]: change string to number

**Review by Flooooooooooorian — APPROVED** [review]

**Pull request merged**